File: lab4b.py
def bubblesort(elm, tamanho):
    
    for i in range(tamanho):
        for j in range(tamanho-1):
            if elm[j] > elm[j+1]:
                t = elm[j]
                elm[j] = elm[j+1]
                elm[j+1] = t
    
    return elm

def mediana(vetor, n):
    if n%2 == 0:
        a = vetor[n//2]
        b = vetor[(n//2)-1]
        med = (a+b)/2
    else:
        med = vetor[n//2]
    
    return med

a = []
t = 0
tamanho = 0

while t != 11:
    print("Adicione um numero:")
    t = float(input("(Digite 11 para encerrar)"))
    if (t >= 0) and (t <= 10):
        tamanho += 1
        a.insert(int(tamanho)-1, t)
    elif t != 11:
        print("\nNumero invalido! Somente permitido numeros entre 0 e 10!\n")

print("\nVetor digitado: ", a)


a = bubblesort(a,int(tamanho))

print("\nSequencia ordenada: ", a)

m = mediana(a, int(tamanho))

print("\nMediana: ", m)

        

File: lab4b.c
#include <stdio.h>
#include <stdlib.h>

float* BubbleSort(float* , int );
float Mediana (float* , int );

int main ()
{
    float *a;
    float elm, mediana;
    int size = 0, i;

    a = (float*)malloc(sizeof(float));

    do {
        printf("Digite um numero entre 0 e 10:\n");
        printf("Encerrar => 11\n ");
        scanf("%f", &elm);
        if(elm >= 0 && elm <= 10)
        {
        size++;
        a = (float*)realloc(a, size*sizeof(float));
        a[size - 1] = elm;
        }
        else if(elm != 11)
            printf("\nNumero invalido! Somente permitido numeros entre 0 e 10!\n\n");

    }
    while(elm != 11);

    printf("\nSequencia digitada: ");

    for (i=0; i < size; i++)
    {
        printf("%.2f\t ", a[i]);
    }
    printf("\n");

    a = BubbleSort(a, size);

    printf("\nSequencia ordenada: ");

    for (i=0; i < size; i++)
    {
        printf("%.2f ", a[i]);
    }
    printf("\n");

    mediana = Mediana(a,size);

    printf("\nMediana: %.2f\n", mediana);


    return 0;
}

float* BubbleSort(float* vetor, int size)
{
    int i, j;
    float t;

    for (i=0; i < size;i++)
    {
        for (j=0; j < size - 1;j++)
        {
            if(vetor[j] > vetor [j+1])
            {
                t = vetor[j];
                vetor[j] = vetor[j+1];
                vetor[j+1] = t;
            }
        }
    }
    return vetor;
}

float Mediana (float* ptr, int size)
{
    int i, j;
    float mediana, a, b;

    if (size%2 == 0)
    {
        a = ptr[size/2];
        b = ptr[(size/2)-1];
        mediana = (a + b)/2;
    }
    else
        mediana = ptr[(size/2)];

    return mediana;
}